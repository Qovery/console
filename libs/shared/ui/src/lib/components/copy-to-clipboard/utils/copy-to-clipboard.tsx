export function copyToClipboard(str: string): void {
  navigator.clipboard.writeText(str)
}
