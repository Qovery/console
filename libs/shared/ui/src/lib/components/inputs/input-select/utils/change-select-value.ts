import { act, fireEvent, getByRole, getByTestId } from '@testing-library/react'

export function simulateSpaceKeyClick(element: Element) {
  fireEvent.keyDown(element, { key: ' ' })
  fireEvent.keyUp(element, { key: ' ' })
}

export async function changeSelectValueByLabel(
  baseElement: HTMLElement,
  selectTestId: string,
  value: string
): Promise<void> {
  simulateSpaceKeyClick(getByTestId(getByTestId(baseElement, selectTestId), 'input-select-button'))

  const itemToSelect = getByRole(baseElement, 'option', {
    name: value,
  })
  await act(() => {
    fireEvent.click(itemToSelect)
  })
}
