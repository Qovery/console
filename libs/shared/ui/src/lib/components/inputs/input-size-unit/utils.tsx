export const inputSizeUnitRules = (maxSize?: number, minSize = 0) => ({
  required: 'Please enter a size.',
  validate: (value: number) => (maxSize ? value <= maxSize : undefined),
  max: maxSize,
  min: minSize,
  pattern: {
    value: /^[0-9]+$/,
    message: 'Please enter a number.',
  },
})
