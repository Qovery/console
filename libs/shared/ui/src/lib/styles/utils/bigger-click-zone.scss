.bigger-click-zone {
  position: relative;

  &:after {
    content: '';
    position: absolute;
    inset: -4px;
    // background: rgba(red, .3);
    display: block;
  }
}
