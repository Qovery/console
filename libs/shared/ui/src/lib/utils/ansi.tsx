// eslint-disable-next-line @typescript-eslint/no-restricted-imports
import AnsiPrimitive from 'ansi-to-react'
import { type ComponentProps } from 'react'
import { twMerge } from '@qovery/shared/util-js'

interface AnsiProps extends ComponentProps<typeof AnsiPrimitive> {}

export function Ansi({ className, children, ...props }: AnsiProps) {
  return (
    <AnsiPrimitive className={twMerge('code-ansi', className)} linkify {...props}>
      {children}
    </AnsiPrimitive>
  )
}
