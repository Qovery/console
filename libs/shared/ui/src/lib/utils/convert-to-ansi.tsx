import Ansi from 'ansi-to-react'
import { ReactNode } from 'react'

export const convertToAnsi = (content = ''): ReactNode => {
  return (
    <Ansi className="code-ansi" linkify>
      {content}
    </Ansi>
  )
}
