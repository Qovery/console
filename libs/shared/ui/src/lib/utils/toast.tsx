import { Toast, toast as toastAction } from 'react-hot-toast'
import { ToastContent } from '../components/toast/toast'

export enum ToastEnum {
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
  WARNING = 'WARNING',
}

export const toast = (
  status: ToastEnum,
  title: string,
  description?: string,
  callback?: () => void,
  iconAction?: string,
  labelAction?: string,
  externalLink?: string
) => {
  return toastAction.success((options: Toast) =>
    ToastContent(status, options, title, description, callback, iconAction, labelAction, externalLink)
  )
}

export default toast
