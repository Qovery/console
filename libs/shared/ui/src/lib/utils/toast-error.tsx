import { type SerializedError } from '@qovery/shared/utils'
import toast, { ToastEnum } from './toast'

export function toastError(error: SerializedError | Error, title?: string, message?: string): void {
  toast(ToastEnum.ERROR, title || error.name || 'Error', message || error.message || 'No message found')
}
