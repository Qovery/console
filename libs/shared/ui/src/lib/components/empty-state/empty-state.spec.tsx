import { findByText, queryByTestId, queryByText } from '@testing-library/react'
import { render } from '__tests__/utils/setup-jest'
import EmptyState, { EmptyStateProps } from './empty-state'

const props: EmptyStateProps = {
  title: 'No Storage are set',
}

describe('EmptyState', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<EmptyState {...props} />)
    expect(baseElement).toBeTruthy()
  })

  it('should render the title', () => {
    const { baseElement } = render(<EmptyState {...props} />)
    findByText(baseElement, 'No Storage are set')
  })

  it('should not render the description block if no description provided', async () => {
    const { baseElement } = render(<EmptyState {...props} />)
    const paragraph = queryByTestId(baseElement, 'placeholder-settings-description')
    expect(paragraph).toBeNull()
  })

  it('should render the description block if no description provided', async () => {
    props.description = 'Need help? You may find these links useful'
    const { baseElement } = render(<EmptyState {...props} />)
    const paragraph = queryByText(baseElement, 'Need help? You may find these links useful')
    expect(paragraph).not.toBeNull()
  })
})
