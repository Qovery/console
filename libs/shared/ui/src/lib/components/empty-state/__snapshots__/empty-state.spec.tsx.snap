// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-grow items-center justify-center"
    >
      <div
        class="m-auto mt-12 flex w-[420px] flex-col items-center justify-center text-center"
      >
        <img
          alt="Event placeholder"
          class="user-none pointer-events-none w-[170px] dark:hidden"
          src="/assets/images/event-placeholder-light.svg"
        />
        <img
          alt="Event placeholder"
          class="user-none pointer-events-none hidden w-[170px] dark:block"
          src="/assets/images/event-placeholder-dark.svg"
        />
        <p
          class="text-neutral-400 dark:text-neutral-300"
        >
          No Storage are set
        </p>
        <p
          class="mt-1 text-sm text-neutral-350"
        >
          Need help? You may find these links useful
        </p>
      </div>
    </div>
  </div>
</body>
`;
